### ISSUE TITLE
can't build on macos

### ISSUE THREAD

**Issue by lukad** [raylib-rs/raylib-rs#102]
I'm on macos 15.1.1 on an M2 machine and I can't get raylib-rs to compile.
It appears that there is some mismatch of compile targets:

```
  thread 'main' panicked at /Users/luka/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bindgen-0.70.1/lib.rs:892:13:
  assertion `left == right` failed: "arm64-apple-darwin" "aarch64-apple-darwin"
```

I don't fully understand what's going on here, `clang -print-target-triple` gives me `arm64-apple-darwin24.1.0` but `cat target/debug/build/bindgen-7d370858a199688f/out/host-target.txt` shows `aarch64-apple-darwin`. Is this a bug in bindgen or some misconfiguration in raylib-sys's build.rs?

Would be greatly appreciated if someone can help me build raylib!

Here is the build error with `RUST_BACKTRACE=full`

```
error: failed to run custom build command for `raylib-sys v5.0.1 (/Users/luka/code/raylib-rs/raylib-sys)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_DEV_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-4126ecee31da6f96/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build.rs
  cargo:rerun-if-changed=./binding/binding.h
  CMAKE_TOOLCHAIN_FILE_aarch64-apple-darwin = None
  CMAKE_TOOLCHAIN_FILE_aarch64_apple_darwin = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_aarch64-apple-darwin = None
  CMAKE_GENERATOR_aarch64_apple_darwin = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_aarch64-apple-darwin = None
  CMAKE_PREFIX_PATH_aarch64_apple_darwin = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_aarch64-apple-darwin = None
  CMAKE_aarch64_apple_darwin = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/build" && CMAKE_PREFIX_PATH="" LC_ALL="C" "cmake" "/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/raylib" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DCMAKE_BUILD_TYPE=Debug" "-DBUILD_EXAMPLES=OFF" "-DCMAKE_BUILD_TYPE=Debug" "-DSUPPORT_BUSY_WAIT_LOOP=OFF" "-DSUPPORT_FILEFORMAT_JPG=ON" "-DRAYMATH_STATIC_INLINE=ON" "-DUSE_WAYLAND=OFF" "-DOPENGL_VERSION=OFF" "-DPLATFORM=Desktop" "-DCMAKE_INSTALL_PREFIX=/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx15.1" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx15.1" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-macosx15.1" "-DCMAKE_ASM_COMPILER=/usr/bin/cc"
  -- Testing if -Werror=pointer-arith can be used -- compiles
  -- Testing if -Werror=implicit-function-declaration can be used -- compiles
  -- Testing if -fno-strict-aliasing can be used -- compiles
  -- Using raylib's GLFW
  -- Including Cocoa support
  -- Audio Backend: miniaudio
  -- Building raylib static library
  -- Generated build type: Debug
  -- Compiling with the flags:
  --   PLATFORM=PLATFORM_DESKTOP
  --   GRAPHICS=GRAPHICS_API_OPENGL_33
  -- Configuring done (0.0s)
  -- Generating done (0.0s)
  -- Build files have been written to: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/build
  running: cd "/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/build" && LC_ALL="C" MAKEFLAGS="-j --jobserver-fds=8,9 --jobserver-auth=8,9" "cmake" "--build" "/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/build" "--target" "install" "--config" "Debug"
  [ 73%] Built target glfw
  [100%] Built target raylib
  Install the project...
  -- Install configuration: "Debug"
  -- Installing: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/lib/libraylib.a
  -- Up-to-date: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/include/raylib.h
  -- Up-to-date: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/include/rlgl.h
  -- Up-to-date: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/include/raymath.h
  -- Up-to-date: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/lib/pkgconfig/raylib.pc
  -- Up-to-date: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/lib/cmake/raylib/raylib-config-version.cmake
  -- Up-to-date: /Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/lib/cmake/raylib/raylib-config.cmake
  cargo:root=/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out
  cargo:rustc-link-search=native=/Users/luka/code/raylib-rs/target/debug/build/raylib-sys-f940292ab14ab324/out/lib

  --- stderr
  CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 3.10 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.


  CMake Warning (dev) at cmake/LibraryConfigurations.cmake:22 (MESSAGE):
    OpenGL is deprecated starting with macOS 10.14 (Mojave)!
  Call Stack (most recent call first):
    src/CMakeLists.txt:48 (include)
  This warning is for project developers.  Use -Wno-dev to suppress it.

  gmake: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
  thread 'main' panicked at /Users/luka/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bindgen-0.70.1/lib.rs:892:13:
  assertion `left == right` failed: "arm64-apple-darwin" "aarch64-apple-darwin"
    left: 4
   right: 8
  stack backtrace:
     0:        0x100dab32c - std::backtrace_rs::backtrace::libunwind::trace::h6285d60e8a2c022a
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/../../backtrace/src/backtrace/libunwind.rs:116:5
     1:        0x100dab32c - std::backtrace_rs::backtrace::trace_unsynchronized::h21fc036d4b8dd7fe
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
     2:        0x100dab32c - std::sys::backtrace::_print_fmt::h69654d1f9cd065f2
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/sys/backtrace.rs:66:9
     3:        0x100dab32c - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h7adefaad4a31afc0
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/sys/backtrace.rs:39:26
     4:        0x100dc9da4 - core::fmt::rt::Argument::fmt::h42151196bb7af094
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/fmt/rt.rs:177:76
     5:        0x100dc9da4 - core::fmt::write::h381c0b0ce6ab972a
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/fmt/mod.rs:1186:21
     6:        0x100da7d10 - std::io::Write::write_fmt::h75af97148630d8d3
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/io/mod.rs:1839:15
     7:        0x100dab1e0 - std::sys::backtrace::BacktraceLock::print::h8baf33e22611de71
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/sys/backtrace.rs:42:9
     8:        0x100dac6d0 - std::panicking::default_hook::{{closure}}::h369c7295ef58c5b1
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:268:22
     9:        0x100dac514 - std::panicking::default_hook::h50746358288a9d6a
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:295:9
    10:        0x100dace84 - std::panicking::rust_panic_with_hook::h7d795911432661cb
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:801:13
    11:        0x100dacb38 - std::panicking::begin_panic_handler::{{closure}}::h36f15310ecbde379
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:674:13
    12:        0x100dab7f0 - std::sys::backtrace::__rust_end_short_backtrace::heed121414170e0c7
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/sys/backtrace.rs:170:18
    13:        0x100dac7f0 - rust_begin_unwind
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:665:5
    14:        0x100de1c80 - core::panicking::panic_fmt::h17b1b80ec02ffd19
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/panicking.rs:74:14
    15:        0x100de1fcc - core::panicking::assert_failed_inner::h13f6fb8338d443dd
    16:        0x100ddfb7c - core::panicking::assert_failed::h12458362c3fbdb5a
    17:        0x1009f6bb8 - bindgen::Bindings::generate::h2069e4ef3c94157d
    18:        0x1009f5238 - bindgen::Builder::generate::h01809cc320717283
    19:        0x1008eb410 - build_script_build::gen_bindings::h017d0151d6500707
    20:        0x1008ebdf4 - build_script_build::main::h4ccf095350042796
    21:        0x1008ec978 - core::ops::function::FnOnce::call_once::h2abcb711a5dfc46a
    22:        0x1008e68c0 - std::sys::backtrace::__rust_begin_short_backtrace::h129379d5d4484afb
    23:        0x1008e5af8 - std::rt::lang_start::{{closure}}::hbf759f17b907f82b
    24:        0x100da450c - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h1f2e4bfdaa1e9f55
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/ops/function.rs:284:13
    25:        0x100da450c - std::panicking::try::do_call::hf5f5fcb9152f95d7
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:557:40
    26:        0x100da450c - std::panicking::try::hc37bea4722ea5011
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:520:19
    27:        0x100da450c - std::panic::catch_unwind::h59a2106aa502b0e8
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panic.rs:358:14
    28:        0x100da450c - std::rt::lang_start_internal::{{closure}}::h130ec563acce7cfd
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/rt.rs:174:48
    29:        0x100da450c - std::panicking::try::do_call::h406987cb4ab94a01
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:557:40
    30:        0x100da450c - std::panicking::try::h3095cbba23c315df
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panicking.rs:520:19
    31:        0x100da450c - std::panic::catch_unwind::hc35a49412f9c6ccf
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/panic.rs:358:14
    32:        0x100da450c - std::rt::lang_start_internal::h77891a2543177e4b
                                 at /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/std/src/rt.rs:174:20
    33:        0x1008e5ad0 - std::rt::lang_start::h474c2bc0aeba80f9
    34:        0x1008ec84c - _main
```

**Comment by IoIxD** [comment]
The discussion at https://github.com/rust-lang/cc-rs/issues/1278 suggests that this is a Rust upstream issue. It's not really anything we can solve (we actually don't even make mention of Mac OS other then to tell CMake to build for Mac OS).

A friend who has an M4 mac is telling me that the version of clang installed from XCode works for him to build `raylib-rs`. Can you confirm?

**Comment by lukad** [comment]
Thanks for the reply. I am  already using the clang install from Xcode command command line tools. I made sure that that I'm not using anything installed from Homebrew. I'll have a look at the issue you mentioned.

**Comment by IoIxD** [comment]
Are you able to use C -> Rust FFI at all, with or without bindgen? Can you set up a sample project with either?

**Comment by lukad** [comment]
I checked and unfortunately, I can’t use bindgen at all. The [tutorial](https://rust-lang.github.io/rust-bindgen/tutorial-0.html) also fails with the same error.  I tested with both the latest clang version installed through llvm via homebrew and with the clang version that ships with xcode. However, I can successfully use FFI via libc.

I’m not sure where the issue lies, but it seems like something in bindgen might be broken. Given my limited experience with bindgen, etc., I’ve decided to move on for now.

Thanks a for your help, though!
